packages/@aws-cdk/aws-cloudwatch/test/cross-environment.test.ts

```diff
@@ -112,6 +113,36 @@ describe('cross environment', () => {
 
 
     });
+
+    test('metric attached to stack3 will render in stack1 ', () => {
```

Comment:
Can you add a test that uses the method of setting the account property on the metric instead of using attachTo? There should be a unit test for each way that it is possible to use the feature.

Reply:
I added an additional test that uses the `account` property, but I do want to push back on this a bit (or ask for clarification). Under the hood, `metric.AttachTo()` simply updates the `metric` object with the `account` property, so I see it as a convenience method and not an additional way to use this feature. To me, it seems like the second test duplicates the first. Does this still warrant an additional unit test?

Reply:
I suppose it is somewhat of a grey area. But I still think the extra test is helpful. Your reasoning for the additional unit test being unnecessary is based on an implementation detail. What if that changes? In this particular scenario, it doesn't seem likely. But in general, unit tests should test the API in all ways that a customer could use it, and make sure that the correct behaviour/result happens in all of them. So that if someone accidentally breaks something by changing an implementation detail, then the unit tests will catch it.